repos/kaleido/js/src/mermaid/render.js

```diff
@@ -0,0 +1,66 @@
+/* global Mermaid:false */
+
+const semver = require('semver')
+const hasOwn = require('object.hasown')
+
+if (!Object.hasOwn) {
+	hasOwn.shim();
+}
+
+/**
+ * @param {object} info : info object
+ *  - data
+ *  - format
+ *  - width
+ *  - height
+ *  - scale
+ */
+function render (info) {
+
+  // Rename info.data to info.markdown
+  info.markdown = info.data
+  delete info.data;
+  
+  let errorCode = 0
+  let result = null
+  let errorMsg = null
+  let pdfBgColor = null
+  
+  const done = () => {
+    if (errorCode !== 0 && !errorMsg) {
+      errorMsg = cst.statusMsg[errorCode]
+    }
+
+    return {
+      code: errorCode,
+      message: errorMsg,
+      pdfBgColor,
+      format: info.format,
+      result,
+      width: info.width,
+      height: info.height,
+      scale: info.scale,
+    }
+  }
+
+  let promise
+
+  // TODO create different rendering call for v<10 and v>=10 of mermaidjs ?
```

Comment:
This should be just different handling of result, plain svg string vs { svg }, right? I agree we should handle it

Reply:
Yes, but I need to read changelog a bit more. I planned to handle that in the next PR.